hilti/src/rt/tests/regexp.cc

```diff
@@ -83,6 +83,25 @@ TEST_SUITE_END();
 
 TEST_SUITE_BEGIN("MatchState");
 
+TEST_CASE("advance") {
+    CHECK_EQ(RegExp("123", regexp::Flags()).tokenMatcher().advance("123"_b, false), std::make_tuple(-1, 3));
```

Comment:
hmm, shouldn't this return `(1, 3)`?

Reply:
I agree, but I am unsure what is happening here. We put a non-final `Byte` into the matcher which almost immediately forwards it into `jrx_regex_partial`. I believe jrx should return something else here as we (as in `rt::RegExp`) cannot tell if the pattern contained e.g., wildcards towards the end.

This test documents the existing behavior. I added a `TODO` to look into that independently.